packages/bsky/src/views/index.ts

```diff
@@ -1049,45 +1113,56 @@ export class Views {
     } else if (parsedUri.collection === ids.AppBskyFeedGenerator) {
       const view = this.feedGenerator(uri, state)
       if (!view) return this.embedNotFound(uri)
-      view.$type = 'app.bsky.feed.defs#generatorView'
-      return this.recordEmbedWrapper(view, withTypeTag)
+      return this.recordEmbedWrapper(
+        { ...view, $type: 'app.bsky.feed.defs#generatorView' },
+        withTypeTag,
+      )
```

Comment:
This is a good example of the benefit of this change. Before, because `$type` was optional, the type system did not force the `$type` property to be present in unions.

Now, because it is no longer optional in unions, not specifying the `$type` here results in a type error.